modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -327,14 +323,40 @@ def ip(self, materialize=True):
         str
             IP address of the node that holds the data.
         """
-        if self._ip_cache is None:
-            if len(self.call_queue):
-                self.drain_call_queue()
-            else:
-                self._ip_cache = self.apply(lambda df: pandas.DataFrame([]))._ip_cache
-        if materialize and isinstance(self._ip_cache, ObjectIDType):
-            self._ip_cache = RayWrapper.materialize(self._ip_cache)
-        return self._ip_cache
+        if (ip := self._ip_cache) is None:
+            self.drain_call_queue()
+        if materialize and isinstance(ip, ObjectIDType):
+            self._ip_cache = ip = RayWrapper.materialize(ip)
+        return ip
+
+    @property
+    def _data(self):  # noqa: GL08
+        self.drain_call_queue()
+        return self._data_ref
```

Comment:
`.data` field is used at higher levels of logic, let's put a hint type here for the values that can be returned.

Reply:
Done.